src/ast.rs

```diff
@@ -113,14 +113,16 @@ struct Parser<'i> {
 impl<'i> Parser<'i> {
   /// Book = ("@" Name "=" Net)*
   fn parse_book(&mut self) -> Result<Book, String> {
-    let mut book = BTreeMap::new();
-    while self.consume("@").is_ok() {
-      let name = self.parse_name()?;
-      self.consume("=")?;
-      let net = self.parse_net()?;
-      book.insert(name, net);
-    }
-    Ok(Book { nets: book })
+    stacker::maybe_grow(1024 * 32, 1024 * 1024, move || {
```

Comment:
repeating these constants everywhere isn't great; it'd be better if they were named constants (or even better, a function in `util` that calls stacker with these arguments)